### ISSUE TITLE
Problems in Evaluating and Training

### ISSUE THREAD

**Issue by fenfaxiaomao888** [dariopavllo/convmesh#3]
Hi Dario,

This is really a good work. Thanks for sharing. But I met some problems when running your code.

First, I set up the full environment including the datasets as your guideline, and I did generate pseudo-ground-truth data before evaluating. But I got the same wrong results below when evaluating on cub dataset with Class or Text condition.

FID (training set): 310.52
Texture-only FID (training set): 310.52
Mesh-only FID (training set): 310.52
FID (validation set): 309.70
Texture-only FID (validation set): 309.70
Mesh-only FID (validation set): 309.70

Second, when I run the training, I found that all masks generated by the Discriminator contains all zero, which results in nan GAN loss. 

Could you please give me some hints to solve these problems?

Thank you.

Best,
F

**Comment by dariopavllo** [comment]
Hi,

There is definitely something wrong with the setup! My guess is that you're trying to evaluate the pretrained models, right?
Do you also get those results if you clone a clean repository and test the pretrained models directly (without setting up the pseudo-g.t. dataset)?

Can you provide information on your system (GPU, CUDA version, OS, Python version, PyTorch version)?

**Comment by fenfaxiaomao888** [comment]
Yes, I think there is something wrong with my setup. 

I have cloned a clean repository and tested the pretrained models directly. And I got the same result.
FID (training set): 310.52
FID (validation set): 309.70

I will recheck my environment and chenge PyTorch version to 1.6x. (NOW GPU-GeForce GTX 1080, CUDA-10.0, Python-3.6.2 and PyTorch-1.7.1)

Thank you.

**Comment by fenfaxiaomao888** [comment]
I finally decided to use PyTorch 1.2 and it works.

Thanks for your help.

**Comment by dariopavllo** [comment]
Good to hear!

Btw, I tried out the code on PyTorch 1.7.1 (pre-built version with CUDA 11) and it works just fine. It's hard to say what could have gone wrong in your case... My guess is some version mismatch between the CUDA version of nvcc and the CUDA version of PyTorch. If you are building PyTorch yourself, it could also be an issue with your build, or the fact that CUDA 10.0 is not officially supported by PyTorch 1.7.1.
